buildtools/AWS.Deploy.Tools.CI/AWS.Deploy.Tools.CI/CodeBuildFunctions.cs

```diff
@@ -0,0 +1,84 @@
+    /// A list of IAM roles, representing the test runner accounts, is passed to the function. 
+    /// These roles are assumed and used to check if the CodeBuild CI project in the test runner account is currently running any jobs.
+    /// </summary>
+    /// <returns>The function will return the IAM role of the account that is not running any CodeBuild CI jobs.</returns>
+    /// <exception cref="ArgumentNullException">If the input passed to the function is invalid.</exception>
+    /// <exception cref="Exception">If no test runner account is available.</exception>
+    /// <exception cref="Exception">If a CodeBuild CI project is not found in the test runner account.</exception>
+    [LambdaFunction(Name = "GetAvailableTestRunner", Policies = "@AWSDeployToolsCIGetAvailableTestRunnerLambdaAssumeRolePolicy")]
+    public async Task<string> GetAvailableTestRunner([FromBody] GetAvailableTestRunnerInput input)
+    {
+        if (string.IsNullOrEmpty(input.Roles))
+            throw new ArgumentNullException(nameof(input.Roles));
+        if (string.IsNullOrEmpty(input.ProjectName))
+            throw new ArgumentNullException(nameof(input.ProjectName));
+        if (string.IsNullOrEmpty(input.Branch))
+            throw new ArgumentNullException(nameof(input.Branch));
+
+        var roles = input.Roles.Split(",").Select(x => x.Trim()).ToList();
+
+        if (!roles.Any())
+            throw new ArgumentNullException(nameof(input.Roles));
+
+        foreach (var role in roles)
+        {
+            var assumeRoleResponse = await new AmazonSecurityTokenServiceClient().AssumeRoleAsync(new AssumeRoleRequest
+            {
+                RoleArn = role,
+                RoleSessionName = "DeployToolTestRunner"
+            });
+
+            var stsClient = new AmazonSecurityTokenServiceClient(assumeRoleResponse.Credentials);
+            var callerIdentity = await stsClient.GetCallerIdentityAsync(new GetCallerIdentityRequest());
+
+            var codeBuildClient = new AmazonCodeBuildClient(assumeRoleResponse.Credentials);
+
+            var batchGetProjectsResponse = await codeBuildClient.BatchGetProjectsAsync(new BatchGetProjectsRequest { Names = new List<string> { input.ProjectName } });
+            if (!batchGetProjectsResponse.Projects.Any())
+                throw new Exception($"Could not find any project with the name '{input.ProjectName}' in account '{callerIdentity.Account}'.");
+            var project = batchGetProjectsResponse.Projects.First();
+
+            var listBuildsForProjectResponse = await codeBuildClient.ListBuildsForProjectAsync(new ListBuildsForProjectRequest { ProjectName = project.Name });
+
+            var runningBuilds = 0;
+            if (listBuildsForProjectResponse.Ids.Any())
+            {
+                var latestBuilds = listBuildsForProjectResponse.Ids.Take(20).ToList();
```

Comment:
I wouldn't say guaranteed, but its highly likely. I did it this way because the SDK doesn't have an API for us to retrieve the running builds only. I would needs to retrieve all the jobs that exist in a project and loop through them to get the running builds, which is something I didn't want to do.

Reply:
Okay, it does default sort order to descending - https://docs.aws.amazon.com/codebuild/latest/APIReference/API_ListBuildsForProject.html#API_ListBuildsForProject_RequestSyntax

Would feel a little safer if we set that explicitly, but not a blocker.

Reply:
![image](https://user-images.githubusercontent.com/53088140/212389961-7337be42-0d47-4ae2-b27f-95c5b286dc09.png)
Setting the sort order will result in an error if the project has more than 100 builds. So I'll leave it as is which default to Descending order.

Reply:
Interesting....okay with it as written then...